--- Makefile.orig	2017-04-26 14:08:43.000000000 +0200
+++ Makefile	2017-07-13 16:43:28.000000000 +0200
@@ -33,6 +33,12 @@
 LOCAL_MKDIRS:=$(shell mkdir -p $(LOCAL_TMP) $(LOCAL_LIB))
 CXX_COMMON:=-I$(LOCAL_INCLUDE) $(CXX_COMMON)

+LOCAL_ARCH:=$(shell uname)
+ifeq ($(LOCAL_ARCH),Darwin)
+   F77LIB:=$(shell gfortran -print-file-name=libgfortran.dylib 2>&1 | grep /)
+   F77LIBDIR:=-L$(dir $(F77LIB))
+endif
+
 # PYTHIA.
 OBJECTS=$(patsubst $(LOCAL_SRC)/%.cc,$(LOCAL_TMP)/%.o,\
 	$(wildcard $(LOCAL_SRC)/*.cc))
@@ -113,7 +119,7 @@
 $(LOCAL_LIB)/libpythia8lhapdf5.so: $(LOCAL_TMP)/LHAPDF5Plugin.o\
 	$(LOCAL_LIB)/libpythia8.a
 	$(CXX) $^ -o $@ $(CXX_COMMON) $(CXX_SHARED) $(CXX_SONAME)$(notdir $@)\
-	 -L$(LHAPDF5_LIB) -Wl,-rpath,$(LHAPDF5_LIB) -lLHAPDF -lgfortran
+	 -L$(LHAPDF5_LIB) -Wl,-rpath,$(LHAPDF5_LIB) -lLHAPDF $(F77LIBDIR) -lgfortran
 $(LOCAL_LIB)/libpythia8lhapdf6.so: $(LOCAL_TMP)/LHAPDF6Plugin.o\
 	$(LOCAL_LIB)/libpythia8.a
 	$(CXX) $^ -o $@ $(CXX_COMMON) $(CXX_SHARED) $(CXX_SONAME)$(notdir $@)\
