--- TAUOLA/platform/make.inc.in.org	2017-05-18 17:26:31.000000000 +0200
+++ TAUOLA/platform/make.inc.in	2017-05-18 17:27:10.000000000 +0200
@@ -7,9 +7,9 @@
 F77FLAGS = -fPIC -fno-automatic -fno-backslash -ffixed-line-length-132
 FFLAGS   = @FFLAGS@ $(F77FLAGS)

-LD       = @F77@
+LD       = @CC@
 LDFLAGS  =
-SOFLAGS  = -shared
+SOFLAGS  = -shared -undefined dynamic_lookup

 RANLIB   = @RANLIB@
 AR       = ar
