--- montecarlo/pythia8/src/TPythia8Decayer.cxx.org	2017-05-19 13:25:40.000000000 +0200
+++ montecarlo/pythia8/src/TPythia8Decayer.cxx	2017-05-19 13:27:56.000000000 +0200
@@ -34,7 +34,7 @@
 ///constructor
 
 TPythia8Decayer::TPythia8Decayer():
-  fPythia8(new TPythia8()),
+  fPythia8(TPythia8::Instance()),
   fDebug(0)
 {
    fPythia8->Pythia8()->readString("SoftQCD:elastic = on");
@@ -46,6 +46,7 @@
 
 void TPythia8Decayer::Init()
 {
+   fPythia8->Pythia8()->init();
 }
 

@@ -59,6 +60,7 @@
    fPythia8->Pythia8()->particleData.mayDecay(idPart,kTRUE);
    fPythia8->Pythia8()->moreDecays();
    if (fDebug > 0) fPythia8->EventListing();
+   fPythia8->Pythia8()->particleData.mayDecay(idPart,kFALSE);
 }
