--- montecarlo/pythia8/src/TPythia8Decayer.cxx
+++ montecarlo/pythia8/src/TPythia8Decayer.cxx
@@ -26,7 +26,7 @@ ClassImp(TPythia8Decayer)
 ///constructor
 
 TPythia8Decayer::TPythia8Decayer():
-  fPythia8(new TPythia8()),
+  fPythia8(TPythia8::Instance()),
   fDebug(0)
 {
    fPythia8->Pythia8()->readString("SoftQCD:elastic = on");
@@ -38,6 +38,7 @@ TPythia8Decayer::TPythia8Decayer():
 
 void TPythia8Decayer::Init()
 {
+fPythia8->Pythia8()->init();
 }
 
 ////////////////////////////////////////////////////////////////////////////////
@@ -51,6 +52,7 @@ void TPythia8Decayer::Decay(Int_t pdg, TLorentzVector* p)
    fPythia8->Pythia8()->particleData.mayDecay(idPart,kTRUE);
    fPythia8->Pythia8()->moreDecays();
    if (fDebug > 0) fPythia8->EventListing();
+   fPythia8->Pythia8()->particleData.mayDecay(idPart,kFALSE);
 }
 
 ////////////////////////////////////////////////////////////////////////////////
@@ -113,4 +115,3 @@ void TPythia8Decayer::ClearEvent()
 {
    fPythia8->Pythia8()->event.clear();
 }
